crates/binjs_io/src/entropy/write/mod.rs

```diff
@@ -0,0 +1,507 @@
+                })?;
+
+            // FIXME: For extensibility purposes, if the value is not in the dictionary, we should
+            // add it to the prelude and a relevant content stream.
+
+            // 2. This gives us an index (`symbol.index`) and a probability distribution
+            // (`symbol.distribution`). Use them to write the probability at bit-level.
+            let mut borrow = symbol.distribution
+                .borrow_mut();
+            $me.writer.symbol(symbol.index.into(), borrow.deref_mut())
+                .map_err(TokenWriterError::WriteError)?;
+
+            // 3. Also, update statistics
+            $me.content_opus_lengths
+                .$info
+                .symbol(symbol.index.into(), borrow.deref_mut())
+                .map_err(TokenWriterError::WriteError)?;
+            $me.content_instances
+                .$info += Into::<Instances>::into(1);
+            Ok(())
+        }
+    }
+}
+
+/// Add a user-extensible symbol to one of the content streams.
+/// If the symbol is not part of either the static dictionary
+/// or the prelude dictionary, it is added to the latter.
+///
+/// This macro is designed for simple values whose binary representation
+/// may be concatenated without loss of information. For string-like values
+/// that need additional info, see `string_content_stream`.
+///
+///
+/// Used instead of a method as we need to generality wrt the field name.
+///
+/// Usage:
+/// `simple_content_stream!(self, name_of_the_indexed_table, name_of_the_stream, value_to_encode, "Description, used for debugging")`
+macro_rules! simple_content_stream {
+    ( $me: ident, $dictionary:ident, $out:ident, $writer: ident, $value: expr, $description: expr ) => {
+        if let Fetch::Miss(_) = shared_content_stream!($me, $dictionary, $out, $value, $description) {
+            // The value does not appear either in the static dictionary or in the prelude dictionary.
+            // Add it to the latter.
+            $me.prelude_streams.$out.increment();
+            $me.prelude_streams.$out.$writer(*$value)
+                .map_err(TokenWriterError::WriteError)?;
+        }
+    }
+}
+
+/// Add a user-extensible symbol to one of the content streams.
+/// If the symbol is not part of either the static dictionary
+/// or the prelude dictionary, it is added to the latter.
+///
+/// This macro is designed for string-like values, which may not
+/// be simply concatenated to obtain a dictionary. For string-like
+/// values, in addition to concatenation, we also record a list of
+/// lengths, which we use during decoding to extract individual
+/// strings.
+///
+/// Used instead of a method as we need to generality wrt the field name.
+///
+/// Usage:
+/// `simple_content_stream!(self, name_of_the_indexed_table, name_of_the_string_prelude_stream, name_of_the_string_length_prelude_stream, value_to_encode, "Description, used for debugging")`
```

Comment:
Usage doesn't match.

Reply:
How so?

Reply:
this is `string_content_stream` macro, but the Usage is about `simple_content_stream`

Reply:
Oops, thanks.